src/Orleans/LogConsistency/ILogConsistencyProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿
+using System;
+using System.Net;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Orleans.Providers;
+using Orleans.Runtime;
+using Orleans.Core;
+using Orleans.Storage;
+
+namespace Orleans.LogConsistency
+{
+    /// <summary>
+    /// Interface to be implemented for a log consistency provider.
+    /// </summary>
+    public interface ILogConsistencyProvider : IProvider, ILogViewAdaptorFactory
+    {
+        /// <summary>Gets the TraceLogger used by this log-consistency provider.</summary>
+        Logger Log { get; }
+
+    }
+
+
+    /// <summary>
+    /// Interface to be implemented for a log-view adaptor factory
+    /// </summary>
+    public interface ILogViewAdaptorFactory  
+    {
+        /// <summary> Returns true if a storage provider is required for constructing adaptors. </summary>
+        bool UsesStorageProvider { get; }
```

Comment:
`Orleans.EventSourcing.CustomStorage.LogConsistencyProvider` returns false, because it does not use a storage provider for interfacing with storage (rather it uses user-supplied read and update functions).